compiler/rustc_interface/src/queries.rs

```diff
@@ -154,13 +154,16 @@ impl Linker {
     }
 
     pub fn link(self, sess: &Session, codegen_backend: &dyn CodegenBackend) -> Result<()> {
-        let (codegen_results, work_products) =
-            codegen_backend.join_codegen(self.ongoing_codegen, sess, &self.output_filenames);
+        let (codegen_results, work_products) = sess.time("finish_ongoing_codegen", || {
+            codegen_backend.join_codegen(self.ongoing_codegen, sess, &self.output_filenames)
+        });
 
         if let Some(guar) = sess.dcx().has_errors() {
             return Err(guar);
         }
 
+        let _timer = sess.timer("link");
```

Comment:
heads up @lqd @nnethercote timings of the "link" section do not include the timing of "finish_ongoing_codegen" anymore.

Reply:
I can revert it if you want, but it felt wrong that link contains finish_ongoing_codegen as the latter doesn't do any linking tasks. It is just a continuation of codegen_crate.

Reply:
I think this change is correct, I just want them to be aware of it if any of their tests suddenly have a major change in timings

Reply:
Is [this heuristic](https://github.com/rust-lang/rustc-perf/blob/2b7adc98bf1735699bbe7c0e18769eae11a3d12f/site/src/self_profile.rs#L254) still correct after this change? It seems to me that it should be. I specifically avoided looking at `link` before, and used `link_crate` instead, because `link` included `finish_ongoing_codegen`.

Reply:
I think it is still correct.